File: Sys/Store.java
package Sys;

import clients.Client;
import clients.Order;
import products.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.Scanner;

public class Store {
    static Scanner sc = new Scanner(System.in);
    private ArrayList<Makeup> makeups = new ArrayList<>();
    private ArrayList<Cleaning> cleaningProducts = new ArrayList<>();
    private ArrayList<Electrodomestico> electrodomesticos = new ArrayList<>();
    private ArrayList<Food> foods = new ArrayList<>();
    private ArrayList<Product> products = new ArrayList<>();
    private ArrayList<Client> clients = new ArrayList<>();
    private ArrayList<Order> orders = new ArrayList<>();

    public ArrayList<Client> getClients() {
        return clients;
    }

    public void registerProduct(){
        String types = "comidaelectrodomesticomaquillajelimpieza";
        System.out.println("\nIngrese el nombre del producto: ");
        String name = sc.next();

        System.out.println("\nIngrese el precio: ");
        double price = sc.nextDouble();

        System.out.println("\nCual sera el stock inicial: ");
        int stock = sc.nextInt();

        System.out.println("\nQue dia se importo el producto: ");
        String importDate = sc.next();

        System.out.println("\nQue tipo de producto sera: (Comida / Electrodomestico / Limpieza / Maquillaje)");
        String type = sc.next();

        while (!types.contains(type)){
            System.out.println("\nTipo no reconocido por el sistema, ingrese uno valido: ");
            type = sc.next();
        }

        switch (type.toLowerCase()){
            case "comida" :
                System.out.println("\nCuales son las calorias del producto: ");
                int calories = sc.nextInt();
                Food food = new Food(name, importDate, price, stock, calories);
                foods.add(food);
                products.add(food);
                break;
            case "electrodomestico":
                System.out.println("\nIngrese la energia que gasta: (En Watts)");
                double energyUsed = sc.nextDouble();
                Electrodomestico electrodomestico = new Electrodomestico(name, importDate, price, stock, energyUsed);
                electrodomesticos.add(electrodomestico);
                products.add(electrodomestico);
                break;
            case "limpieza" :
                System.out.println("\nQue aroma tiene el producto: ");
                String aroma = sc.next();
                Cleaning cleaning = new Cleaning(name, importDate, price, stock, aroma);
                cleaningProducts.add(cleaning);
                products.add(cleaning);
                break;
            case "maquillaje" :
                System.out.println("\nDe que color es el maquillaje: ");
                String color = sc.next();
                Makeup makeup = new Makeup(name, importDate, price, stock ,color);
                makeups.add(makeup);
                products.add(makeup);
                break;
        }
        System.out.println("\n== SE HA REGISTRADO EL PRODUCTO CORRECTAMENTE ==");
    }

    public void registerClient(){
        System.out.println("Ingrese el nombre del cliente: ");
        String name = sc.next();

        System.out.println("Ingrese el apellido del cliente: ");
        String lastName = sc.next();

        System.out.println("Ingrese la edad del cliente: ");
        int age = sc.nextInt();

        System.out.println("Ingrese la direccion del cliente: ");
        String address = sc.next();

        System.out.println("Cual es el saldo del cliente: ");
        double wallet = sc.nextDouble();

        Client client = new Client(name, lastName, address, age, wallet);
        clients.add(client);
        System.out.println("\n== SE HA REGISTRADO EL CLIENTE CORRECTAMENTE ==");
    }

    public void printMakeup(){
        if (makeups.isEmpty()){
            System.out.println("\n== NO HAY PRODUCTOS EN EL SISTEMA ==");
            return;
        }
        System.out.println("\n== LISTA DE PRODUCTOS DE MAQUILLAJE ==");
        for (Makeup makeup : makeups) {
            System.out.printf("=== PRODUCTO %s: ===\n", makeup.getSeriesNumber());
            System.out.printf("""
                    \tNombre: %s Precio: %f Fecha Importe: %s
                    \tStock: %d Color: %s\s
                    """, makeup.getName(), makeup.getPrice(), makeup.getImportDate(), makeup.getStock(), makeup.getColor());
        }
    }

    public void printFood(){
        if (foods.isEmpty()){
            System.out.println("\n== NO HAY PRODUCTOS EN EL SISTEMA ==");
            return;
        }
        System.out.println("\n== LISTA DE PRODUCTOS DE COMIDA ==");
        for (Food food : foods) {
            System.out.printf("=== PRODUCTO %s: ===\n", food.getSeriesNumber());
            System.out.printf("""
                    \tNombre: %s Precio: %f Fecha Importe: %s
                    \tStock: %d Calorias: %s\s
                    """, food.getName(), food.getPrice(), food.getImportDate(), food.getStock(), food.getCalories());
        }
    }

    public void printCleaning(){
        if (cleaningProducts.isEmpty()){
            System.out.println("\n== NO HAY PRODUCTOS EN EL SISTEMA ==");
            return;
        }
        System.out.println("\n== LISTA DE PRODUCTOS DE LIMPIEZA ==");
        for (Cleaning cleaning : cleaningProducts) {
            System.out.printf("=== PRODUCTO %s: ===\n", cleaning.getSeriesNumber());
            System.out.printf("""
                    \tNombre: %s Precio: %f Fecha Importe: %s
                    \tStock: %d Aroma: %s\s
                    """, cleaning.getName(), cleaning.getPrice(), cleaning.getImportDate(), cleaning.getStock(), cleaning.getAroma());
        }
    }

    public void printElectrodomestico(){
        if (electrodomesticos.isEmpty()){
            System.out.println("\n== NO HAY PRODUCTOS EN EL SISTEMA ==");
            return;
        }
        System.out.println("\n== LISTA DE PRODUCTOS ELECTRODOMESTICOS ==");
        for (Electrodomestico electrodomestico : electrodomesticos) {
            System.out.printf("=== PRODUCTO %s: ===\n", electrodomestico.getSeriesNumber());
            System.out.printf("""
                    \tNombre: %s Precio: %f Fecha Importe: %s
                    \tStock: %d Watts: %s\s
                    """, electrodomestico.getName(), electrodomestico.getPrice(), electrodomestico.getImportDate(), electrodomestico.getStock(), electrodomestico.getEnergyUsed());
        }
    }

    public void printClients(){
        if (clients.isEmpty()){
            System.out.println("\n== NO HAY CLIENTES EN EL SISTEMA ==");
            return;
        }
        System.out.println("\n== LISTA DE CLIENTES ==");
        for (Client client: clients){
            System.out.printf("=== CLIENTE %s: ===\n", client.getId());
            System.out.printf("\tNombre: %s %s Edad: %d Direccion: %s Compras Totales: %d Saldo: %f\n",
                    client.getName(),client.getLastName(),client.getAge(),client.getAddress(),client.getTimesBought(),client.getWallet());
        }
    }

    public void printOrder(){
        if (orders.isEmpty()){
            System.out.println("\n== NO HAY COMPRAS EN EL SISTEMA ==");
            return;
        }
        System.out.println("\n== LISTA DE COMPRAS ==");
        for (Order order : orders){
            System.out.printf("Cliente: %s Precio Total: %f Fecha: %s\n",
                    order.getName(),order.getTotalPrice(),order.getDate());
            order.printProducts();
        }
    }

    public void buyOrder(){
        ArrayList<String> productNames = new ArrayList<>();
        double totalPrice = 0 ;

        if (clients.isEmpty()){
            System.out.println("\n== NO HAY CLIENTES EN EL SISTEMA ==");
            return;
        }

        if (electrodomesticos.isEmpty() && makeups.isEmpty() && cleaningProducts.isEmpty() && foods.isEmpty()){
            System.out.println("\n== NO HAY PRODUCTOS EN EL SISTEMA ==");
            return;
        }

        printClients();
        System.out.println("Quien va a realizar la compra: (Ingrese su ID)");
        String id = sc.next();

        while (!FindID.clientExists(this.clients,id)){
            System.out.println("\nID NO RECONOCIDO, INGRESE OTRO: ");
            id = sc.next();
        }
        Client client = FindID.findClient(clients, id);
        String clientName = STR."\{FindID.findClient(clients, id).getName()} \{FindID.findClient(clients, id).getLastName()}";

        if (!cleaningProducts.isEmpty()){printCleaning();}
        if (!electrodomesticos.isEmpty())printElectrodomestico();
        if (!foods.isEmpty())printFood();
        if (!makeups.isEmpty())printMakeup();

        while (true){
            double price = 0;
            System.out.println("Ingrese el numero de serie de los productos que compro el cliente: (fin para salir)");
            String seriesNumber = sc.next();

            if (seriesNumber.equalsIgnoreCase("fin")){
                break;
            }
            if (FindID.productExists(products, seriesNumber) && client.getWallet() >= FindID.findProduct(products, seriesNumber).getPrice()){
                price = FindID.findProduct(products, seriesNumber).getPrice();
                String productName = FindID.findProduct(products, seriesNumber).getName();
                FindID.findProduct(products, seriesNumber).reduceStock(1);
                productNames.add(productName);
                totalPrice += price;
                client.reduceWallet(price);
                System.out.println("SE AGREGO EL PRODUCTO CORRECTAMENTE\n");
            } else if (client.getWallet() < FindID.findProduct(products, seriesNumber).getPrice()) {
                System.out.println("\nEL CLIENTE NO TIENE SUFICIENTES FONDOS PARA HACER LA COMPRA\n");
            }
        }
        if (productNames.isEmpty()){
            System.out.println("\nNO SE PUDO REGISTRAR LA COMPRA, DEBIDO A QUE NO HAY PRODUCTOS");
            return;
        }
        String date = LocalDate.now().toString();
        Order order = new Order(clientName,productNames,totalPrice,date);
        orders.add(order);
        System.out.println("\n== SE HA REGISTRADO LA COMPRA CORRECTAMENTE ==");
    }

    public void modifyProduct(){
        if (!cleaningProducts.isEmpty()){printCleaning();}
        if (!electrodomesticos.isEmpty())printElectrodomestico();
        if (!foods.isEmpty())printFood();
        if (!makeups.isEmpty())printMakeup();

        System.out.println("Ingrese el numero de serie del producto a modificar: ");
        String seriesNumber = sc.next();

        while (!FindID.productExists(products, seriesNumber)){
            System.out.println("Ingrese el numero de serie del producto a modificar: ");
            seriesNumber = sc.next();
        }

        boolean flag = true;
        Product product = FindID.findProduct(products, seriesNumber);

        while (flag) {
            System.out.println("\nQue desea modificar: ");
            System.out.println("1. Nombre");
            System.out.println("2. Precio");
            System.out.println("3. Stock");
            System.out.println("4. Salir");

            int opt = sc.nextInt();

            switch (opt){
                case 1 -> product.setName();
                case 2 -> product.setPrice();
                case 3 -> product.changeStock();
                case 4 -> flag = false;
                default -> System.out.println("\nOpcion ingresada incorrecta!");
            }
        }
    }

    public void modifyClient(){
        printClients();

        System.out.println("Ingrese el ID del cliente a modificar: ");
        String id = sc.next();

        while (!FindID.clientExists(clients, id)){
            System.out.println("\nIngrese el ID del cliente a modificar: ");
            id = sc.next();
        }

        boolean flag = true;
        Client client = FindID.findClient(clients, id);

        while (flag) {
            System.out.println("\nQue desea modificar: ");
            System.out.println("1. Nombre");
            System.out.println("2. Saldo");
            System.out.println("3. Numero de Compras");
            System.out.println("4. Salir");

            int opt = sc.nextInt();

            switch (opt){
                case 1 -> client.setName();
                case 2 -> client.setWallet();
                case 3 -> client.setTimesBought();
                case 4 -> flag = false;
                default -> System.out.println("\nOpcion ingresada incorrecta!");
            }
        }
    }

    public void deleteClient(){
        printClients();

        System.out.println("Ingrese el ID del cliente a eliminar: ");
        String id = sc.next();

        while (!FindID.clientExists(clients, id)){
            System.out.println("\nIngrese el ID del cliente a eliminar: ");
            id = sc.next();
        }
        Client client = FindID.findClient(clients,id);
        System.out.printf("\n== SE HA ELIMINADO A %s CORRECTAMENTE ==",
                client.getName().toUpperCase());
        clients.remove(client);
    }

    public void deleteProduct(){
        String types = "comidaelectrodomesticomaquillajelimpieza";

        System.out.println("\nQue tipo de producto desea eliminar: ");
        String type = sc.next();

        while (!types.contains(type.toLowerCase())){
            System.out.println("\nTIPO INCORRECTO, INGRESE OTRO");
            System.out.println("\nQue tipo de producto desea eliminar: ");
            type = sc.next();
        }

        switch (type.toLowerCase()){
            case "comida" -> deleteFood();
            case "electrodomestico" -> deleteElectrodomestico();
            case "maquillaje" -> deleteMakeup();
            case "limpieza" -> deleteCleaning();
        }
    }

    public void deleteFood(){
        printFood();
        System.out.println("Ingrese el numero de serie del producto a modificar: ");
        String seriesNumber = sc.next();

        while (!FindID.productExists(products, seriesNumber)){
            System.out.println("\nNUMERO INCORRECTO");
            System.out.println("\nIngrese el numero de serie del producto a modificar: ");
            seriesNumber = sc.next();
        }

        Product product = FindID.findProduct(products, seriesNumber);
        products.remove(product);
        foods.remove(product);
        System.out.println("\n== SE HA ELIMINADO CORRECTAMENTE EL PRODUCTO ==");
    }

    public void deleteCleaning(){
        printCleaning();
        System.out.println("Ingrese el numero de serie del producto a modificar: ");
        String seriesNumber = sc.next();

        while (!FindID.productExists(products, seriesNumber)){
            System.out.println("\nNUMERO INCORRECTO");
            System.out.println("\nIngrese el numero de serie del producto a modificar: ");
            seriesNumber = sc.next();
        }

        Product product = FindID.findProduct(products, seriesNumber);
        products.remove(product);
        cleaningProducts.remove(product);
        System.out.println("\n== SE HA ELIMINADO CORRECTAMENTE EL PRODUCTO ==");
    }

    public void deleteElectrodomestico(){
        printElectrodomestico();
        System.out.println("Ingrese el numero de serie del producto a modificar: ");
        String seriesNumber = sc.next();

        while (!FindID.productExists(products, seriesNumber)){
            System.out.println("\nNUMERO INCORRECTO");
            System.out.println("\nIngrese el numero de serie del producto a modificar: ");
            seriesNumber = sc.next();
        }

        Product product = FindID.findProduct(products, seriesNumber);
        products.remove(product);
        electrodomesticos.remove(product);
        System.out.println("\n== SE HA ELIMINADO CORRECTAMENTE EL PRODUCTO ==");
    }

    public void deleteMakeup(){
        printMakeup();
        System.out.println("Ingrese el numero de serie del producto a modificar: ");
        String seriesNumber = sc.next();

        while (!FindID.productExists(products, seriesNumber)){
            System.out.println("\nNUMERO INCORRECTO");
            System.out.println("\nIngrese el numero de serie del producto a modificar: ");
            seriesNumber = sc.next();
        }

        Product product = FindID.findProduct(products, seriesNumber);
        products.remove(product);
        makeups.remove(product);
        System.out.println("\n== SE HA ELIMINADO CORRECTAMENTE EL PRODUCTO ==");
    }

    public ArrayList<Makeup> getMakeups() {
        return makeups;
    }

    public ArrayList<Cleaning> getCleaningProducts() {
        return cleaningProducts;
    }

    public ArrayList<Electrodomestico> getElectrodomesticos() {
        return electrodomesticos;
    }

    public ArrayList<Food> getFoods() {
        return foods;
    }

    public ArrayList<Product> getProducts() {
        return products;
    }
}

File: store.py
import datetime
from cleaning import Cleaning
from food import Food
from makeup import Makeup
from electrodomestico import Electrodomestico
from client import Client
from findID import FindID
from order import Order

class Store:
    sc = input
    def __init__(self):
        self.makeups = []
        self.cleaningProducts = []
        self.electrodomesticos = []
        self.foods = []
        self.products = []
        self.clients = []
        self.orders = []

    def getClients(self):
        return self.clients

    def registerProduct(self):
        types = "comidaelectrodomesticomaquillajelimpieza"
        print("\nIngrese el nombre del producto: ")
        name = input()

        print("\nIngrese el precio: ")
        price = float(input())

        print("\nCual sera el stock inicial: ")
        stock = int(input())

        print("\nQue dia se importo el producto: ")
        importDate = input()

        print("\nQue tipo de producto sera: (Comida / Electrodomestico / Limpieza / Maquillaje)")
        type = input()

        while type.lower() not in types:
            print("\nTipo no reconocido por el sistema, ingrese uno valido: ")
            type = input()

        if type.lower() == "comida":
            print("\nCuales son las calorias del producto: ")
            calories = int(input())
            food = Food(name, importDate, price, stock, calories)
            self.foods.append(food)
            self.products.append(food)
        elif type.lower() == "electrodomestico":
            print("\nIngrese la energia que gasta: (En Watts)")
            energyUsed = float(input())
            electrodomestico = Electrodomestico(name, importDate, price, stock, energyUsed)
            self.electrodomesticos.append(electrodomestico)
            self.products.append(electrodomestico)
        elif type.lower() == "limpieza":
            print("\nQue aroma tiene el producto: ")
            aroma = input()
            cleaning = Cleaning(name, importDate, price, stock, aroma)
            self.cleaningProducts.append(cleaning)
            self.products.append(cleaning)
        elif type.lower() == "maquillaje":
            print("\nDe que color es el maquillaje: ")
            color = input()
            makeup = Makeup(name, importDate, price, stock, color)
            self.makeups.append(makeup)
            self.products.append(makeup)
        print("\n== SE HA REGISTRADO EL PRODUCTO CORRECTAMENTE ==")

    def registerClient(self):
        print("Ingrese el nombre del cliente: ")
        name = input()

        print("Ingrese el apellido del cliente: ")
        lastName = input()

        print("Ingrese la edad del cliente: ")
        age = int(input())

        print("Ingrese la direccion del cliente: ")
        address = input()

        print("Cual es el saldo del cliente: ")
        wallet = float(input())

        client = Client(name, lastName, address, age, wallet)
        self.clients.append(client)
        print("\n== SE HA REGISTRADO EL CLIENTE CORRECTAMENTE ==")

    def printMakeup(self):
        if not self.makeups:
            print("\n== NO HAY PRODUCTOS EN EL SISTEMA ==")
            return
        print("\n== LISTA DE PRODUCTOS DE MAQUILLAJE ==")
        for makeup in self.makeups:
            print(f"=== PRODUCTO {makeup.getSeriesNumber()}: ===")
            print(f"\tNombre: {makeup.getName()} Precio: {makeup.getPrice()} Fecha Importe: {makeup.getImportDate()}")
            print(f"\tStock: {makeup.getStock()} Color: {makeup.getColor()}")

    def printFood(self):
        if not self.foods:
            print("\n== NO HAY PRODUCTOS EN EL SISTEMA ==")
            return
        print("\n== LISTA DE PRODUCTOS DE COMIDA ==")
        for food in self.foods:
            print(f"=== PRODUCTO {food.getSeriesNumber()}: ===")
            print(f"\tNombre: {food.getName()} Precio: {food.getPrice()} Fecha Importe: {food.getImportDate()}")
            print(f"\tStock: {food.getStock()} Calorias: {food.getCalories()}")

    def printCleaning(self):
        if not self.cleaningProducts:
            print("\n== NO HAY PRODUCTOS EN EL SISTEMA ==")
            return
        print("\n== LISTA DE PRODUCTOS DE LIMPIEZA ==")
        for cleaning in self.cleaningProducts:
            print(f"=== PRODUCTO {cleaning.getSeriesNumber()}: ===")
            print(f"\tNombre: {cleaning.getName()} Precio: {cleaning.getPrice()} Fecha Importe: {cleaning.getImportDate()}")
            print(f"\tStock: {cleaning.getStock()} Aroma: {cleaning.getAroma()}")

    def printElectrodomestico(self):
        if not self.electrodomesticos:
            print("\n== NO HAY PRODUCTOS EN EL SISTEMA ==")
            return
        print("\n== LISTA DE PRODUCTOS ELECTRODOMESTICOS ==")
        for electrodomestico in self.electrodomesticos:
            print(f"=== PRODUCTO {electrodomestico.getSeriesNumber()}: ===")
            print(f"\tNombre: {electrodomestico.getName()} Precio: {electrodomestico.getPrice()} Fecha Importe: {electrodomestico.getImportDate()}")
            print(f"\tStock: {electrodomestico.getStock()} Watts: {electrodomestico.getEnergyUsed()}")

    def printClients(self):
        if not self.clients:
            print("\n== NO HAY CLIENTES EN EL SISTEMA ==")
            return
        print("\n== LISTA DE CLIENTES ==")
        for client in self.clients:
            print(f"=== CLIENTE {client.getId()}: ===")
            print(f"\tNombre: {client.getName()} {client.getLastName()} Edad: {client.getAge()} Direccion: {client.getAddress()} Compras Totales: {client.getTimesBought()} Saldo: {client.getWallet()}")

    def printOrder(self):
        if not self.orders:
            print("\n== NO HAY COMPRAS EN EL SISTEMA ==")
            return
        print("\n== LISTA DE COMPRAS ==")
        for order in self.orders:
            print(f"Cliente: {order.getName()} Precio Total: {order.getTotalPrice()} Fecha: {order.getDate()}")
            order.printProducts()

    def buyOrder(self):
        productNames = []
        totalPrice = 0

        if not self.clients:
            print("\n== NO HAY CLIENTES EN EL SISTEMA ==")
            return

        if not self.electrodomesticos and not self.makeups and not self.cleaningProducts and not self.foods:
            print("\n== NO HAY PRODUCTOS EN EL SISTEMA ==")
            return

        self.printClients()
        print("Quien va a realizar la compra: (Ingrese su ID)")
        id = input()

        while not FindID.clientExists(self.clients, id):
            print("\nID NO RECONOCIDO, INGRESE OTRO: ")
            id = input()
        client = FindID.findClient(self.clients, id)
        clientName = f"{FindID.findClient(self.clients, id).getName()} {FindID.findClient(self.clients, id).getLastName()}"

        if self.cleaningProducts:
            self.printCleaning()
        if self.electrodomesticos:
            self.printElectrodomestico()
        if self.foods:
            self.printFood()
        if self.makeups:
            self.printMakeup()

        while True:
            price = 0
            print("Ingrese el numero de serie de los productos que compro el cliente: (fin para salir)")
            seriesNumber = input()

            if seriesNumber.lower() == "fin":
                break
            if FindID.productExists(self.products, seriesNumber) and client.getWallet() >= FindID.findProduct(self.products, seriesNumber).getPrice():
                price = FindID.findProduct(self.products, seriesNumber).getPrice()
                productName = FindID.findProduct(self.products, seriesNumber).getName()
                FindID.findProduct(self.products, seriesNumber).reduceStock(1)
                productNames.append(productName)
                totalPrice += price
                client.reduceWallet(price)
                print("SE AGREGO EL PRODUCTO CORRECTAMENTE\n")
            elif client.getWallet() < FindID.findProduct(self.products, seriesNumber).getPrice():
                print("\nEL CLIENTE NO TIENE SUFICIENTES FONDOS PARA HACER LA COMPRA\n")
        if not productNames:
            print("\nNO SE PUDO REGISTRAR LA COMPRA, DEBIDO A QUE NO HAY PRODUCTOS")
            return
        date = datetime.date.today().strftime("%Y-%m-%d")
        order = Order(clientName, productNames, totalPrice, date)
        self.orders.append(order)
        print("\n== SE HA REGISTRADO LA COMPRA CORRECTAMENTE ==")

    def modifyProduct(self):
        if self.cleaningProducts:
            self.printCleaning()
        if self.electrodomesticos:
            self.printElectrodomestico()
        if self.foods:
            self.printFood()
        if self.makeups:
            self.printMakeup()

        print("Ingrese el numero de serie del producto a modificar: ")
        seriesNumber = input()

        while not FindID.productExists(self.products, seriesNumber):
            print("Ingrese el numero de serie del producto a modificar: ")
            seriesNumber = input()

        flag = True
        product = FindID.findProduct(self.products, seriesNumber)

        while flag:
            print("\nQue desea modificar: ")
            print("1. Nombre")
            print("2. Precio")
            print("3. Stock")
            print("4. Salir")

            opt = int(input())

            if opt == 1:
                product.setName()
            elif opt == 2:
                product.setPrice()
            elif opt == 3:
                product.changeStock()
            elif opt == 4:
                flag = False
            else:
                print("\nOpcion ingresada incorrecta!")

    def modifyClient(self):
        self.printClients()

        print("Ingrese el ID del cliente a modificar: ")
        id = input()

        while not FindID.clientExists(self.clients, id):
            print("\nIngrese el ID del cliente a modificar: ")
            id = input()

        flag = True
        client = FindID.findClient(self.clients, id)

        while flag:
            print("\nQue desea modificar: ")
            print("1. Nombre")
            print("2. Saldo")
            print("3. Numero de Compras")
            print("4. Salir")

            opt = int(input())

            if opt == 1:
                client.setName()
            elif opt == 2:
                client.setWallet()
            elif opt == 3:
                client.setTimesBought()
            elif opt == 4:
                flag = False
            else:
                print("\nOpcion ingresada incorrecta!")

    def deleteClient(self):
        self.printClients()

        print("Ingrese el ID del cliente a eliminar: ")
        id = input()

        while not FindID.clientExists(self.clients, id):
            print("\nIngrese el ID del cliente a eliminar: ")
            id = input()
        client = FindID.findClient(self.clients, id)
        print(f"\n== SE HA ELIMINADO A {client.getName().upper()} CORRECTAMENTE ==")
        self.clients.remove(client)

    def deleteProduct(self):
        types = "comidaelectrodomesticomaquillajelimpieza"

        print("\nQue tipo de producto desea eliminar: ")
        type = input()

        while type.lower() not in types:
            print("\nTIPO INCORRECTO, INGRESE OTRO")
            print("\nQue tipo de producto desea eliminar: ")
            type = input()

        if type.lower() == "comida":
            self.deleteFood()
        elif type.lower() == "electrodomestico":
            self.deleteElectrodomestico()
        elif type.lower() == "maquillaje":
            self.deleteMakeup()
        elif type.lower() == "limpieza":
            self.deleteCleaning()

    def deleteFood(self):
        self.printFood()
        print("Ingrese el numero de serie del producto a modificar: ")
        seriesNumber = input()

        while not FindID.productExists(self.products, seriesNumber):
            print("\nNUMERO INCORRECTO")
            print("\nIngrese el numero de serie del producto a modificar: ")
            seriesNumber = input()

        product = FindID.findProduct(self.products, seriesNumber)
        self.products.remove(product)
        self.foods.remove(product)
        print("\n== SE HA ELIMINADO CORRECTAMENTE EL PRODUCTO ==")

    def deleteCleaning(self):
        self.printCleaning()
        print("Ingrese el numero de serie del producto a modificar: ")
        seriesNumber = input()

        while not FindID.productExists(self.products, seriesNumber):
            print("\nNUMERO INCORRECTO")
            print("\nIngrese el numero de serie del producto a modificar: ")
            seriesNumber = input()

        product = FindID.findProduct(self.products, seriesNumber)
        self.products.remove(product)
        self.cleaningProducts.remove(product)
        print("\n== SE HA ELIMINADO CORRECTAMENTE EL PRODUCTO ==")

    def deleteElectrodomestico(self):
        self.printElectrodomestico()
        print("Ingrese el numero de serie del producto a modificar: ")
        seriesNumber = input()

        while not FindID.productExists(self.products, seriesNumber):
            print("\nNUMERO INCORRECTO")
            print("\nIngrese el numero de serie del producto a modificar: ")
            seriesNumber = input()

        product = FindID.findProduct(self.products, seriesNumber)
        self.products.remove(product)
        self.electrodomesticos.remove(product)
        print("\n== SE HA ELIMINADO CORRECTAMENTE EL PRODUCTO ==")

    def deleteMakeup(self):
        self.printMakeup()
        print("Ingrese el numero de serie del producto a modificar: ")
        seriesNumber = input()

        while not FindID.productExists(self.products, seriesNumber):
            print("\nNUMERO INCORRECTO")
            print("\nIngrese el numero de serie del producto a modificar: ")
            seriesNumber = input()

        product = FindID.findProduct(self.products, seriesNumber)
        self.products.remove(product)
        self.makeups.remove(product)
        print("\n== SE HA ELIMINADO CORRECTAMENTE EL PRODUCTO ==")

    def getMakeups(self):
        return self.makeups

    def getCleaningProducts(self):
        return self.cleaningProducts

    def getElectrodomesticos(self):
        return self.electrodomesticos

    def getFoods(self):
        return self.foods

    def getProducts(self):
        return self.products


